src/main/java/com/aws/greengrass/mqtt/bridge/MessageBridge.java

```diff
@@ -146,8 +156,21 @@ private <T extends Message> void handleMessage(TopicMapping.TopicType sourceType
     private <T extends Message> void publishMessage(MessageClient<T> client, String topic, Message message)
             throws MessageClientException {
         T msg = client.convertMessage(message);
-        msg = (T) msg.newFromMessageWithTopic(topic);
-        client.publish(msg);
+
+        if (isRetainAsPublished(topic) && msg instanceof MqttMessage) {
+            MqttMessage mqttMsg = (MqttMessage) msg;
+            mqttMsg.setRetain(true);
```

Comment:
why are you setting retain? You need to read retain first so that it is set to the same value as it was published with, right?

Reply:
I'm reading the retain setting from the mqtt5 route options in the `isRetainAsPublished` call, and then if the retain flag is set there then I update the message accordingly. I just updated this to use the builder instead of a setter

Reply:
The retain flag is not set by the route though.
The retain flag is in the incoming message.

Reply:
Makes sense now, I had the wrong understanding of the purpose of the route. Now I'm setting the retain flag based off of the incoming message when the route has retain as published enabled